rust/main/chains/hyperlane-ethereum/src/contracts/mailbox.rs

```diff
@@ -465,7 +464,9 @@ impl<M: Middleware + 'static> BatchSimulation<M> {
         self,
         cache: Arc<Mutex<EthereumMailboxCache>>,
     ) -> ChainResult<BatchResult> {
-        if let Some(submittable_batch) = self.call {
+        if let Some(mut submittable_batch) = self.call {
+            let estimated = multicall::estimate(submittable_batch.call, self.successful).await?;
```

Comment:
could this be run in parallel to the `fill_tx_gas_params` in `submit`?

Reply:
Function `fill_tx_gas_params` uses gas limit estimate which we got from `multicall::estimate`. So, we cannot.